version: "3"

x-user-configs: &pg-config-env
  POSTGRES_PORT: ${POSTGRES_PORT:-5432}
  POSTGRES_USER: ${POSTGRES_USER:-postgres}
  POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-postgres}
  POSTGRES_DB: ${POSTGRES_DB:-user}
  TZ: Asia/Shanghai

services:
  user-api:
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
      - "8081:9090"
    depends_on:
      user-postgres:
        condition: service_healthy
    environment:
      POSTGRES_HOST: ${POSTGRES_HOST:-postgres}
      <<: *pg-config-env
    networks:
      - another-mentor-network
    volumes:
      - ./logs:/app/logs
      - ./etc:/app/etc
    command: ["./user", "-f", "etc/user-api.docker.yaml"]
    restart: always

  user-postgres:
    image: postgres:17-alpine
    environment:
      <<: *pg-config-env
    ports:
      - "5432:5432"
    volumes:
      - user_data:/var/lib/postgresql/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - another-mentor-network
    restart: always
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  redis:
    image: redis:7.4.2-alpine3.21
    ports:
      - "6379:6379"
    networks:
      - another-mentor-network
    restart: always
    command: ["redis-server", "--appendonly", "yes"]

networks:
  another-mentor-network:
    external: true
  